lldb/source/Expression/DWARFExpression.cpp

```diff
@@ -2153,26 +2152,42 @@ bool DWARFExpression::Evaluate(
             }
             break;
 
-          case Value::ValueType::FileAddress:
-          case Value::ValueType::HostAddress:
-            if (error_ptr) {
-              lldb::addr_t addr = curr_piece_source_value.GetScalar().ULongLong(
-                  LLDB_INVALID_ADDRESS);
+          case Value::ValueType::FileAddress: {
```

Comment:
I'm wondering if it would be possible to use a single implementation for these two. Target::ReadMemory will call Process::ReadMemory if it was a process available (and the `force_live_memory` precisely so that one can require this behavior).